entity_event/context_loader.py

```diff
@@ -0,0 +1,187 @@
+"""
+A module for loading contexts using context hints.
+"""
+from collections import defaultdict
+
+from django.db.models import Q
+from django.db.models.loading import get_model
+from manager_utils import id_dict
+
+from entity_event.models import ContextRenderer
+
+
+def get_context_hints_per_source(context_renderers):
+    """
+    Given a list of context renderers, return a dictionary of context hints per source.
+    """
+    # Merge the context render hints for each source as there can be multiple context hints for
+    # sources depending on the render target. Merging them together involves combining select
+    # and prefetch related hints for each context renderer
+    context_hints_per_source = defaultdict(lambda: defaultdict(lambda: {
+        'app_name': None,
+        'model_name': None,
+        'select_related': set(),
+        'prefetch_related': set(),
+    }))
+    for cr in context_renderers:
+        for key, hints in cr.context_hints.items() if cr.context_hints else []:
+            for source in cr.get_sources():
+                context_hints_per_source[source][key]['app_name'] = hints['app_name']
+                context_hints_per_source[source][key]['model_name'] = hints['model_name']
```

Comment:
They will combine select and prefetch relateds if different sources define different context hints for the same key. If they define different models for the same key, that is just an error anyways and would be problematic